### PR TITLE
Nima and Reactive implementation of OIDC provider separated

### LINKED ISSUE #5395: Nima: OIDC now contains Nima and SE integration
OIDC Support in security now has both Reactive and Níma implementation in the same module - needs to be refactored

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #5395

### CODE REVIEW

File: examples/nima/protocols/src/main/java/io/helidon/examples/nima/protocols/ProtocolsMain.java [hunk 1]
```diff
@@ -109,11 +110,14 @@ public void onClose(WsSession session, int status, String reason) {
     }
 
     private static KeyConfig privateKey() {
-        String password = "helidon";
+        String password = "password";
 
         return KeyConfig.keystoreBuilder()
-                .keystore(Resource.create("certificate.p12"))
-                .keystorePassphrase(password)
+                .keystore(Resource.create("server.p12"))
```
Comment (tomas-langer) [inline examples/nima/protocols/src/main/java/io/helidon/examples/nima/protocols/ProtocolsMain.java:116]: This does not look right - doing the same thing multiple times?

File: microprofile/oidc/src/main/java/io/helidon/microprofile/oidc/OidcCdiExtension.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ private void registerOidcSupport(@Observes @Initialized(ApplicationScoped.class)
             // only configure if security is enabled
             ServerCdiExtension server = bm.getExtension(ServerCdiExtension.class);
 
-            server.serverRoutingBuilder().register(OidcService.create(config));
+            OidcFeature.create(config).setup(server.serverRoutingBuilder());
```
Comment (tomas-langer) [inline microprofile/oidc/src/main/java/io/helidon/microprofile/oidc/OidcCdiExtension.java:48]: This is not correct. Please use `server.serverRutingBuilder.addFeature(OidcFeature.create(config))`

Comment (tomas-langer) [reply microprofile/oidc/src/main/java/io/helidon/microprofile/oidc/OidcCdiExtension.java:48]: Features have `Weight` and are ordered by routing to be executed in the correct order. See `docs-internal/nima-http-features.md` for details

File: security/providers/oidc-reactive/pom.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (tomas-langer) [inline security/providers/oidc-reactive/pom.xml:22]: use `https://` for the location of the namespace (the second URI)

File: security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/DefaultTenantIdProvider.java [hunk 4]
```diff
@@ -19,20 +19,19 @@
...
 
-import jakarta.annotation.Priority;
-
 import static io.helidon.security.providers.oidc.common.spi.TenantConfigFinder.DEFAULT_TENANT_ID;
 
 /**
  * This is the default tenant id provider.
  */
-@Priority(100000)
+@Weight(100000)
```
Comment (tomas-langer) [inline security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/DefaultTenantIdProvider.java:34]: `Priority(100000)` is very low priority. `Weight(100000)` is very high priority (e.g. it is exactly the opposite meaning).
Is this desired? If you want this to be easily overridden by users, use `Weight(1)`

File: security/providers/oidc/src/main/java/module-info.java [hunk 5]
```diff
@@ -22,7 +22,7 @@
  */
 @Feature(value = "OIDC",
         description = "Security provider for Open ID Connect authentication",
-        in = {HelidonFlavor.SE, HelidonFlavor.MP, HelidonFlavor.NIMA},
+        in = {HelidonFlavor.MP, HelidonFlavor.NIMA},
```
Comment (tomas-langer) [inline security/providers/oidc/src/main/java/module-info.java:25]: This is not an MP feature. There is a `microprofile/oidc` module that adds this feature for Helidon MP

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: security/providers/oidc/pom.xml [hunk 6]
```diff
@@ -58,18 +58,6 @@
...
-        </dependency>
-        <dependency>
-            <groupId>io.helidon.reactive.webserver</groupId>
-            <artifactId>helidon-reactive-webserver</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>io.helidon.reactive.webserver</groupId>
-            <artifactId>helidon-reactive-webserver-cors</artifactId>
-        </dependency>
         <dependency>
```
Comment (tomas-langer) [inline security/providers/oidc/pom.xml:61]: Either all dependencies on Níma MUST be as `provided` and `optional` with `requires static` in `module-info.java`, or Níma OIDC integration must be moved to a separate module

Comment (Verdent) [reply security/providers/oidc/pom.xml:61]: Updated

**Review by tomas-langer — APPROVED** [review]

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**